chains/substrate/client/check_metadata_hash_enabled_client.go

```diff
@@ -0,0 +1,214 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/rpc/author"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/scale"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/signature"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/types"
+	"github.com/sygmaprotocol/sygma-core/chains/substrate/connection"
+	"github.com/sygmaprotocol/sygma-core/chains/substrate/events"
+)
+
+type SubstrateCheckMetadataModeEnabledClient struct {
```

Comment:
What's different here exactly? 
Anyway to write so it is not directly copy-pasted?

Reply:
@mpetrun5 I was looking for a way to only change the type of `SignatureOptions` but the new version of it added two [more parameters](https://github.com/centrifuge/go-substrate-rpc-client/blob/add-check-metadata-support-in-extrinsic/types/extrinsic_signature.go#L52-L53), so if only changing the type, it will not be able to convert to the old version and pass to the `submitAndWatchExtrinsic`. That's why instead of changing only `SignatureOptions`, I copied the entire Client implementation with new version so there is no type matching issue, same reason for the connection implementation. Obversely it creates duplications which is bad, good side is that it doesn't touch the current implementation at all and if GSRPC is released with compatibility issue fixing, we can simply remove this duplication. 

To my understanding, this multiple GSRPC support is only a temporary solution until Centrifuge fix on their side. Please correct me if I'm wrong.

Reply:
I can live this I guess, but let's not merge into `main` then. 
Create a separate feature branch and we will install on relayer temporarily as such and then create a separate fix later. 

I would honestly prefer it being inside one client as by their comments I don't foresee them fixing this quickly, but a temporary branch might be ok.